[#17248] Doc: Update documentation for Upgrade and AutoFlag Promotion (#17920)
yugabyte/yugabyte-db: docs/content/stable/admin/yb-admin.md

---

### Upgrade

Refer to [Upgrade a deployment](../../manage/upgrade-deployment/) to learn about how to upgrade a Yugabyte cluster.

#### promote_auto_flags

> [AutoFlags](https://github.com/yugabyte/yugabyte-db/blob/master/architecture/design/auto_flags.md) protect new features that modify the format of data sent over the wire or stored on-disk. Once all Yugabyte processes have been upgraded to the new version, these features can be enabled by promoting their AutoFlags.

Reviewer: ```suggestion
[AutoFlags](https://github.com/yugabyte/yugabyte-db/blob/master/architecture/design/auto_flags.md) protect new features that modify the format of data sent over the wire or stored on-disk. After all YugabyteDB processes have been upgraded to the new version, these features can be enabled by promoting their AutoFlags.
```

**Syntax**

```sh
yb-admin \
    -master_addresses <master-addresses> \
    promote_auto_flags \
    [<max_flags_class> [<promote_non_runtime_flags> [force]]]
```

* *master-addresses*: Comma-separated list of YB-Master hosts and ports. Default value is `localhost:7100`.
* *max_flags_class*: The maximum AutoFlag class to promote. Allowed values are `kLocalVolatile`, `kLocalPersisted`, `kExternal`, `kNewInstallsOnly`. Default value is `kExternal`.
* *promote_non_runtime_flags*: Weather to promote non-runtime flags. Allowed values are `true` and `false`. Default value is `true`.

> * *force*: Forces the generation of a new AutoFlag config and sends it to all Yugabyte processes even if there is no new AutoFlags to promote.

Reviewer: ```suggestion
* *force*: Forces the generation of a new AutoFlag configuration and sends it to all YugabyteDB processes even if there are no new AutoFlags to promote.
```
